bot/cogs/moderation/silence.py

```diff
@@ -0,0 +1,175 @@
+        self.bot.loop.create_task(self._get_instance_vars())
+
+    async def _get_instance_vars(self) -> None:
+        """Get instance variables after they're available to get from the guild."""
+        await self.bot.wait_until_guild_available()
+        guild = self.bot.get_guild(Guild.id)
+        self._verified_role = guild.get_role(Roles.verified)
+        self._mod_alerts_channel = self.bot.get_channel(Channels.mod_alerts)
+        self._mod_log_channel = self.bot.get_channel(Channels.mod_log)
+        self.notifier = SilenceNotifier(self._mod_log_channel)
+
+    @commands.command(aliases=("hush",))
+    async def silence(self, ctx: Context, duration: HushDurationConverter = 10) -> None:
+        """
+        Silence `channel` for `duration` minutes or `forever`.
+
+        Duration is capped at 15 minutes, passing forever makes the silence indefinite.
+        Indefinitely silenced channels get added to a notifier which posts notices every 15 minutes from the start.
+        """
+        log.debug(f"{ctx.author} is silencing channel #{ctx.channel}.")
+        if not await self._silence(ctx.channel, persistent=(duration is None), duration=duration):
+            await ctx.send(f"{Emojis.cross_mark} current channel is already silenced.")
+            return
+        if duration is None:
+            await ctx.send(f"{Emojis.check_mark} silenced current channel indefinitely.")
+            return
+
+        await ctx.send(f"{Emojis.check_mark} silenced current channel for {duration} minute(s).")
+        await asyncio.sleep(duration*60)
+        log.info(f"Unsilencing channel after set delay.")
+        await ctx.invoke(self.unsilence)
+
+    @commands.command(aliases=("unhush",))
+    async def unsilence(self, ctx: Context) -> None:
+        """
+        Unsilence `channel`.
+
+        Unsilence a previously silenced `channel`,
+        remove it from notifier of indefinitely silenced channels and cancel the notifier if empty.
+        """
+        log.debug(f"Unsilencing channel #{ctx.channel} from {ctx.author}'s command.")
+        if await self._unsilence(ctx.channel):
+            await ctx.send(f"{Emojis.check_mark} unsilenced current channel.")
+
+    async def _silence(self, channel: TextChannel, persistent: bool, duration: Optional[int]) -> bool:
+        """
+        Silence `channel` for `self._verified_role`.
+
+        If `persistent` is `True` add `channel` to notifier.
+        `duration` is only used for logging; if None is passed `persistent` should be True to not log None.
+        """
+        current_overwrite = channel.overwrites_for(self._verified_role)
+        if current_overwrite.send_messages is False:
+            log.info(f"Tried to silence channel #{channel} ({channel.id}) but the channel was already silenced.")
+            return False
+        await channel.set_permissions(
+            self._verified_role,
+            overwrite=PermissionOverwrite(**dict(current_overwrite, send_messages=False))
```

Comment:
> The overwrite parameter, if given, must either be None or PermissionOverwrite. **For convenience, you can pass in keyword arguments denoting Permissions attributes.** If this is done, then you cannot mix the keyword arguments with the overwrite parameter.

Passing kwargs is a better option when only interested in setting one permission. I'm pretty sure it will leave the rest of the overrides intact, but I have not tested.

Reply:
Providing kwargs creates a permission overwrite object from them [internally](https://github.com/Rapptz/discord.py/blob/5e6335750819588062790c6b93d604d3bc7c3862/discord/abc.py#L620-L626), so it doesn't keep them.

But realized that we can skip the PermissionOverwrite itself here and just unpack the dict into kwargs.